packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3010,10 +3049,39 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         type: ContainerMessageType,
         contents: any,
         batch: boolean,
+        contentLength: number,
         appData?: any,
     ) {
         this.verifyNotClosed();
         assert(this.connected, 0x259 /* "Container disconnected when trying to submit system message" */);
+
+        if (contentLength > this.runtimeOptions.compressionOptions?.minimumSize) {
+            const originalContentLength = IsoBuffer.from(JSON.stringify(contents)).toString("base64").length;
+
+            const compressionStart = Date.now();
+            const stringifiedContent = JSON.stringify(contents);
+            const contentsAsBuffer = new TextEncoder().encode(stringifiedContent);
+            const compressedContents = compress(contentsAsBuffer);
+            const compressionTime = Date.now() - compressionStart;
+            const compressedContent = IsoBuffer.from(compressedContents).toString("base64");
+
+            this.logger.sendPerformanceEvent({
+                eventName: "compressedOp",
+                compressionTime,
```

Comment:
Please use `duration` instead of compressionTime - it has special handling in telemetry pipeline (we use floor for floating pointer numbers to make nicer reports and reduce content size in telemetry - for most part JS does not gives us precision above 1ms anyway due to various hacks using timing data).